x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/InlineJoin.java

```diff
@@ -129,4 +133,19 @@ protected NodeInfo<Join> info() {
     public Join replaceChildren(LogicalPlan left, LogicalPlan right) {
         return new InlineJoin(source(), left, right, config());
     }
+
+    @Override
+    public List<Attribute> computeOutput(List<Attribute> left, List<Attribute> right) {
+        JoinType joinType = config().type();
+        List<Attribute> output;
+        if (LEFT.equals(joinType)) {
+            List<Attribute> leftOutput = new ArrayList<>(left().output());
+            AttributeSet rightKeys = new AttributeSet(config().rightFields());
+            List<Attribute> leftOutputWithoutMatchFields = leftOutput.stream().filter(attr -> rightKeys.contains(attr) == false).toList();
```

Comment:
Instead of using streams, you could create an empty collection then do a forEach to add elements in there:
```
 List<Attribute> leftOutputWithoutMatchFields = new ArrayList<>()
 leftOutput.forEach(e -> if (rightKeys.contains(e) == false) { leftOutputWIthoutMatchField.add(e)});
```